src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -1801,15 +1801,49 @@ private IEmptyOperation CreateBoundNoOpStatementOperation(BoundNoOpStatement bou
 
         private IConditionalOperation CreateBoundIfStatementOperation(BoundIfStatement boundIfStatement)
         {
-            IOperation condition = Create(boundIfStatement.Condition);
-            IOperation whenTrue = Create(boundIfStatement.Consequence);
-            IOperation? whenFalse = Create(boundIfStatement.AlternativeOpt);
-            bool isRef = false;
-            SyntaxNode syntax = boundIfStatement.Syntax;
-            ITypeSymbol? type = null;
-            ConstantValue? constantValue = null;
-            bool isImplicit = boundIfStatement.WasCompilerGenerated;
-            return new ConditionalOperation(condition, whenTrue, whenFalse, isRef, _semanticModel, syntax, type, constantValue, isImplicit);
+            var stack = ArrayBuilder<BoundIfStatement>.GetInstance();
+
+            IOperation? whenFalse;
+            while (true)
+            {
+                stack.Push(boundIfStatement);
+
+                var alternative = boundIfStatement.AlternativeOpt;
+                if (alternative is null)
+                {
+                    whenFalse = null;
+                    break;
+                }
+
+                if (alternative is BoundIfStatement elseIfStatement)
+                {
+                    boundIfStatement = elseIfStatement;
+                }
+                else
+                {
+                    whenFalse = Create(alternative);
+                    break;
+                }
```

Comment:
Nit: `Create` is null-resileint. Consider simplifying this to just check if `alternative` is a `BoundIfStatement`, and unconditionally calling `Create` and break in the else.